test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlDocumentTest/CorrectlyHandlesOddlySpacedHTMLElements.stree.txt

```diff
@@ -1,43 +1,48 @@
-            SyntaxKind.Text;[div];
-            SyntaxKind.Whitespace;[ ];
-            SyntaxKind.CloseAngle;[>];
-    Tag block - Gen<None> - 17 - (6:0,6)
-        Markup span - Gen<Markup> - [<p] - SpanEditHandler;Accepts:Any - (6:0,6) - Tokens:2
-            SyntaxKind.OpenAngle;[<];
-            SyntaxKind.Text;[p];
-        Markup block - Gen<Attr:class, class = '@(8:0,8),'@(21:0,21)> - 14 - (8:0,8)
-            Markup span - Gen<None> - [ class = '] - SpanEditHandler;Accepts:Any - (8:0,8) - Tokens:6
-                SyntaxKind.Whitespace;[ ];
-                SyntaxKind.Text;[class];
+SyntaxKind.HtmlDocument - [0..39) - FullWidth: 39 - Slots: 1 - [<div ><p class = 'bar'> Foo </p></div >]
+    SyntaxKind.HtmlMarkupBlock - [0..39) - FullWidth: 39 - Slots: 1
+        SyntaxKind.HtmlTagBlock - [0..6) - FullWidth: 6 - Slots: 1
+            SyntaxKind.HtmlTextLiteral - [0..6) - FullWidth: 6 - Slots: 1 - Gen<Markup> - SpanEditHandler;Accepts:Any
+                SyntaxKind.OpenAngle;[<];
+                SyntaxKind.Text;[div];
                 SyntaxKind.Whitespace;[ ];
+                SyntaxKind.CloseAngle;[>];
+        SyntaxKind.HtmlTagBlock - [6..23) - FullWidth: 17 - Slots: 1
+            SyntaxKind.HtmlTextLiteral - [6..8) - FullWidth: 2 - Slots: 1 - Gen<Markup> - SpanEditHandler;Accepts:Any
+                SyntaxKind.OpenAngle;[<];
+                SyntaxKind.Text;[p];
+            SyntaxKind.HtmlAttributeBlock - [8..22) - FullWidth: 14 - Slots: 7
+                SyntaxKind.HtmlTextLiteral - [8..9) - FullWidth: 1 - Slots: 1 - Gen<Markup> - SpanEditHandler;Accepts:Any
+                    SyntaxKind.Whitespace;[ ];
+                SyntaxKind.HtmlTextLiteral - [9..14) - FullWidth: 5 - Slots: 1 - Gen<Markup> - SpanEditHandler;Accepts:Any
+                    SyntaxKind.Text;[class];
+                SyntaxKind.HtmlTextLiteral - [14..15) - FullWidth: 1 - Slots: 1 - Gen<Markup> - SpanEditHandler;Accepts:Any
+                    SyntaxKind.Whitespace;[ ];
                 SyntaxKind.Equals;[=];
-                SyntaxKind.Whitespace;[ ];
-                SyntaxKind.SingleQuote;['];
-            Markup span - Gen<LitAttr:@(18:0,18)> - [bar] - SpanEditHandler;Accepts:Any - (18:0,18) - Tokens:1
-                SyntaxKind.Text;[bar];
-            Markup span - Gen<None> - ['] - SpanEditHandler;Accepts:Any - (21:0,21) - Tokens:1
-                SyntaxKind.SingleQuote;['];
-        Markup span - Gen<Markup> - [>] - SpanEditHandler;Accepts:Any - (22:0,22) - Tokens:1
-            SyntaxKind.CloseAngle;[>];
-    SyntaxKind.HtmlTextLiteral - [ Foo ] - [23..28) - FullWidth: 5 - Slots: 1
-        SyntaxKind.List - [ Foo ] - [23..28) - FullWidth: 5 - Slots: 3
+                SyntaxKind.HtmlTextLiteral - [16..18) - FullWidth: 2 - Slots: 1 - Gen<None> - SpanEditHandler;Accepts:Any
+                    SyntaxKind.Whitespace;[ ];
+                    SyntaxKind.SingleQuote;['];
+                SyntaxKind.HtmlBlock - [18..21) - FullWidth: 3 - Slots: 1
```

Comment:
This may change when I do pass 2 of HtmlParser rewrite but as of now here is what they mean,
`HtmlBlock` is just a holder that contains bunch of other nodes. E.g. Attribute value, where it can be a combination of different kinds of values
`HtmlMarkupBlock` is more conceptually meaningful and acts as a parent in multiple cases

Reply:
Think of `HtmlBlock` as just a wrapper for `SyntaxList<RazorSyntaxNode>`

Reply:
Super confusing, we should change that 😄 (don't care if we do now or later)

Reply:
I agree. I should change the name atleast.